drivers/wifi/simplelink_support.c

```diff
@@ -0,0 +1,599 @@
+/*
+ * Copyright (c) 2018, Texas Instruments Incorporated
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_WIFI_LEVEL
+#define SYS_LOG_DOMAIN "dev/simplelink"
+#include <logging/sys_log.h>
+
+#include <stdlib.h>
+#include <string.h>
+#include <zephyr.h>
+#include <stdint.h>
+
+#include <ti/drivers/net/wifi/simplelink.h>
+#include <CC3220SF_LAUNCHXL.h>
+
+#include "simplelink_support.h"
+
+#define SET_STATUS_BIT(status, bit) {status |= (1 << (bit)); }
+#define CLR_STATUS_BIT(status, bit) {status &= ~(1 << (bit)); }
+#define GET_STATUS_BIT(status, bit) (0 != (status & (1 << (bit))))
+
+#define SL_STOP_TIMEOUT	 (200)
+
+#undef ASSERT_ON_ERROR
+#define ASSERT_ON_ERROR(ret, e) __ASSERT(ret >= 0, e)
+#define DEVICE_ERROR	 ("See \"DEVICE ERRORS CODES\" in SimpleLink errors.h")
+#define WLAN_ERROR	 ("See \"WLAN ERRORS CODES\" in SimpleLink errors.h")
+#define NETAPP_ERROR	 ("See \"NETAPP ERRORS CODES\" in SimpleLink errors.h")
+
+#define CHANNEL_MASK_ALL	    (0x1FFF)
+#define RSSI_TH_MAX		    (-95)
+
+enum status_bits {
+	/* Network Processor is powered up */
+	STATUS_BIT_NWP_INIT = 0,
+	/* The device is connected to the AP */
+	STATUS_BIT_CONNECTION,
+	/* The device has leased IP to any connected client */
+	STATUS_BIT_IP_LEASED,
+	/* The device has acquired an IP */
+	STATUS_BIT_IP_ACQUIRED,
+	/* The device has acquired an IPv6 address */
+	STATUS_BIT_IPV6_ACQUIRED,
+};
+
+struct nwp_status {
+	/* Callback to notify net & wifi mgmt events from SL Event Handlers */
+	simplelink_wifi_cb_t cb;
+
+	/* Status Variables */
+	u32_t status;	/* The state of the NWP */
+	u32_t role;	/* The device's role (STA, P2P or AP) */
+
+	/* Scan results table: */
+	SlWlanNetworkEntry_t net_entries[CONFIG_WIFI_SIMPLELINK_SCAN_COUNT];
+};
+
+/* STA/AP mode state: shared with simplelink.c */
+struct sl_connect_state sl_conn;
+
+/* Network Coprocessor state, including role and connection state: */
+static struct nwp_status nwp;
+
+/* Configure the device to a default state, resetting previous parameters .*/
+static s32_t configure_simplelink(void)
+{
+	u8_t config_opt;
+	u8_t power;
+	s32_t retval = -1;
+	s32_t mode = -1;
+	u32_t if_bitmap = 0;
+	SlWlanScanParamCommand_t scan_default = { 0 };
+	SlWlanRxFilterOperationCommandBuff_t rx_filterid_mask = { { 0 } };
```

Comment:
tiny point, we usually do the invert here: first declaring variable pre-assigned, and then only the lone declaration.

Reply:
OK